adapta/storage/models/aws.py

```diff
@@ -63,15 +75,20 @@ def from_hdfs_path(cls, hdfs_path: str) -> "S3Path":
 
     def to_hdfs_path(self) -> str:
         """
-        Not yet implemented in S3Path
+        Converts the S3Path to an HDFS compatible path.
+        :return: HDFS path
         """
+        if not self.bucket or not self.path:
+            raise ValueError("Bucket and path must be defined")
+
         return f"s3a://{self.bucket}/{self.path}"
 
     def to_delta_rs_path(self) -> str:
         """
-        Not yet implemented in S3Path
+        Converts the S3Path to a Delta Lake compatible path.
+        :return: Delta Lake path
         """
-        raise NotImplementedError
+        return f"s3a://{self.bucket}/{self.path}"
```

Comment:
This is not valid deltars path?

Reply:
I've tested this successfully, but now I'm unsure. I'll test it again and get back to you on this